netwaystev2/src/filter/sortedbuffer.rs

```diff
@@ -0,0 +1,251 @@
+///! A sorted buffer which maintains ordering and uniqueness in the buffer.
+
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+pub struct SequencedMinHeap<T> {
+    heap: BinaryHeap<Reverse<SequencedItem<T>>>,
+}
+
+impl<T> SequencedMinHeap<T> {
+    pub fn new() -> Self {
+        Self {
+            heap: BinaryHeap::new(),
+        }
+    }
+
+    fn contains_sequence_number(&self, sequence: u64) -> bool {
+        for tuple in &self.heap {
+            if tuple.0.0 == sequence {
+                return true
+            }
+        }
+        false
+    }
+
+    pub fn count_contiguous(&self, mut sequence: u64) -> usize {
+        let mut count = 0;
+        for tuple in &self.heap {
+            if tuple.0.0 == sequence {
+                count += 1;
+                sequence += 1;
```

Comment:
Will this wrap correctly without panicking? @Manghi

Reply:
Also, I noticed the tests are failing for this because there is no way to iterate over the heap in the order that `pop()` will happen (without consuming the heap, which would be inefficient). I wonder if we really need this operation since we may not be able to implement it in an efficient way?

Reply:
> without consuming the heap, which would be inefficient
Exactly!

Thinking some more about this, you're absolutely right, we don't *really* need the operation. There aren't any great sorted set libs presently available for our use case anyways.

I think comparing against the `sequence` via `peek()` before a `pop()` in a loop is the right way to go for now. This is corrected in the latest commit.